doc/designs/functions-for-explicitly-fetched-signature-algorithms.md

```diff
@@ -0,0 +1,205 @@
+Functions for explicitly fetched PKEY algorithms
+================================================
+
+Quick background
+----------------
+
+There are several proposed designs that end up revolving around the same
+basic need, explicitly fetched signature algorithms.  The following method
+type is affected by this document:
+
+- `EVP_SIGNATURE`
+
+Public API - Add variants of `EVP_PKEY_CTX` functionality
+---------------------------------------------------------
+
+Through OTC discussions, it's been determined that the most suitable APIs to
+touch is the collection of `EVP_PKEY_` functions.  In this case, it involves
+`EVP_PKEY_sign()`, `EVP_PKEY_verify()`, `EVP_PKEY_verify_recover()` with
+associated functions.  They can be extended to accept an explicitly fetched
+algorithm of the right type, and to be able to process infinite amounts of
+data if the fetched algorithm permits it (typically, algorithms like ED25519
+or RSA-SHA256).
```

Comment:
```suggestion
Through OTC discussions, it's been determined that the most suitable APIs to
touch are the `EVP_PKEY_` functions.
Specifically, `EVP_PKEY_sign()`, `EVP_PKEY_verify()`, `EVP_PKEY_verify_recover()`
and related functions.
They can be extended to accept an explicitly fetched algorithm of the right type,
and to be able to incrementally process indefinite length data streams when the
fetched algorithm permits it (typically, algorithms like ED25519 or RSA-SHA256).
```
Does Ed25519 really support streaming? Or are you thinking of Ed25519ph?
It was my impression that pure Ed25519 required the entire message in memory
and did not support IUF-style (Init, Update, Final) signing.

Reply:
Ah, I should remove ED25519 from there, 'cause indeed, the entire length must be known from the start, which had escaped me when I wrote this passage.